packages/gallery/src/components/gallery/proGallery/galleryContainerExtraNew.js

```diff
@@ -407,32 +403,27 @@ export class GalleryContainer extends React.Component {
     });
   }
 
-  getScrollCssIfNeeded({ domId, items, styleParams, allowPreloading }) {
-    const shouldUseScrollCss = !isSEOMode();
+  getScrollCssIfNeeded({ domId, items, styleParams }) {
+    const shouldUseScrollCss = !isSEOMode() && (isEditMode() || this.state.gotFirstScrollEvent|| this.state.showMoreClickedAtLeastOnce);
     let scrollCss = [];
     if (shouldUseScrollCss) {
       scrollCss = cssScrollHelper.calcScrollCss({
         items,
         isUnknownWidth: dimensionsHelper.isUnknownWidth(),
         styleParams,
         domId,
-        allowPreloading,
       });
     }
-    return (scrollCss && scrollCss.length > 0) ? scrollCss : this.scrollCss;
+    if (scrollCss && scrollCss.length > 0) {
+      if (JSON.stringify(this.state.scrollCss) !== JSON.stringify(scrollCss)) {
+        this.setState({
+          scrollCss
```

Comment:
why do you need to store it on the state?
The two parameters that trigger it are on the state
Saving too much information on the state is slow